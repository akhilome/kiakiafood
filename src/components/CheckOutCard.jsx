import React from 'react';
import PropTypes from 'prop-types';

const CheckOutCard = ({
  id = 0, foodName = '', foodPrice = 0, buttonCallback,
}) => (
  <div data-foodid={id} className="food-card-checkout">
    <div className="food-card-checkout__details">
      <h4>{foodName}</h4>
      <p>
â‚¦
        {foodPrice}
      </p>
    </div>
    <button type="button" onClick={buttonCallback}>
      Remove
    </button>
  </div>
);

CheckOutCard.propTypes = {
  id: PropTypes.number.isRequired,
  foodName: PropTypes.string.isRequired,
  foodPrice: PropTypes.number.isRequired,
  buttonCallback: PropTypes.func.isRequired,
};

export default CheckOutCard;
