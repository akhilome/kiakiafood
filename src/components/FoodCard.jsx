import React from 'react';
import PropTypes from 'prop-types';

const FoodCard = ({
  foodImage, foodName, id, foodPrice, isAdmin,
}) => (
  <div className="food-card">
    <img className="food-image" src={foodImage} alt={foodName} />
    <div className="food-details">
      <div data-foodid={id} className="food-details__legend">
        <h4>{foodName}</h4>
        <p>
â‚¦
          {foodPrice.toLocaleString()}
        </p>
      </div>
      <div className="food-details__action">
        <button type="button">{isAdmin ? 'delete' : 'buy'}</button>
      </div>
    </div>
  </div>
);

FoodCard.propTypes = {
  foodImage: PropTypes.string.isRequired,
  foodName: PropTypes.string.isRequired,
  foodPrice: PropTypes.number.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  id: PropTypes.number.isRequired,
};

export default FoodCard;
