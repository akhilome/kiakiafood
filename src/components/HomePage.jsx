import React from 'react';
import { Link } from 'react-router-dom';
import Nav from './Nav';

const HomePage = () => (
  <div>
    <Nav />
    <section className="hero">
      <h2>Quell Your Hunger in Minutes</h2>
      <Link to="/menu">
        <button type="button" className="btn-primary">
          Get Food ‚ü∂
        </button>
      </Link>
    </section>
  </div>
);

export default HomePage;
