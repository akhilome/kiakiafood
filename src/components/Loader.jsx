import React from 'react';

const Loader = () => (
  <div className="loader">
    <img src="https://i.imgur.com/S39IaLt.gif" alt="loading gif" />
  </div>
);

export default Loader;
