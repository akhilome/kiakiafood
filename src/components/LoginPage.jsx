import React from 'react';
import { Link } from 'react-router-dom';

const LoginPage = () => {
  return (
    <div>
      This is the login page! ğŸ”¥ See <Link to="/signup">signup</Link>
    </div>
  );
};

export default LoginPage;
