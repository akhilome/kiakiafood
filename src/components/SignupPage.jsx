import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { signUpUser } from '../actions';
import Nav from './Nav';

export class SignupPage extends Component {
  state = {
    name: '',
    email: '',
    password: '',
    confirmPassword: '',
    adminSecret: '',
  };

  componentDidMount() {
    const { isLoggedIn, fetching, history } = this.props;
    if (!fetching && isLoggedIn) history.push('/menu');
  }

  onFormSubmit = async (e) => {
    e.preventDefault();
    const {
      name, email, password, confirmPassword, adminSecret,
    } = this.state;
    const { signUpUser: register } = this.props;
    await register({
      name,
      email,
      password,
      confirmPassword,
      adminSecret,
    });
    const { isLoggedIn, fetching, history } = this.props;
    if (!fetching && isLoggedIn) history.push('/menu');
  };

  render() {
    const {
      name, email, password, confirmPassword, adminSecret,
    } = this.state;
    return (
      <Fragment>
        <Nav />
        <div className="wrapper auth">
          <section className="container">
            <h2>Create An Account</h2>

            <form onSubmit={this.onFormSubmit}>
              <input
                type="text"
                name="name"
                placeholder="Your full name"
                value={name}
                onChange={e => this.setState({ name: e.target.value })}
                required
              />
              <input
                type="email"
                name="email"
                placeholder="Your Email Address"
                value={email}
                onChange={e => this.setState({ email: e.target.value })}
                required
              />
              <input
                type="password"
                name="password"
                placeholder="Choose Password"
                value={password}
                onChange={e => this.setState({ password: e.target.value })}
                required
              />
              <input
                type="password"
                name="password-confirm"
                placeholder="Confirm Password"
                value={confirmPassword}
                onChange={e => this.setState({ confirmPassword: e.target.value })}
                required
              />
              <input
                type="password"
                name="admin-secret"
                placeholder="Admin Secret [optional]"
                value={adminSecret}
                onChange={e => this.setState({ adminSecret: e.target.value })}
              />
              <input type="submit" name="submit" value="Sign Up" className="btn-primary" />
            </form>
            <div>
              <p>
                Already have an account?
                {' '}
                <Link to="/login">Sign In</Link>
              </p>
            </div>
          </section>
        </div>
      </Fragment>
    );
  }
}

SignupPage.defaultProps = {
  isLoggedIn: false,
  fetching: false,
};

SignupPage.propTypes = {
  signUpUser: PropTypes.func.isRequired,
  isLoggedIn: PropTypes.bool,
  fetching: PropTypes.bool,
  history: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
};

const mapStateToProps = state => ({
  isLoggedIn: state.user.isLoggedIn,
  fetching: state.fetching.fetching,
});

export default connect(
  mapStateToProps,
  { signUpUser },
)(SignupPage);
