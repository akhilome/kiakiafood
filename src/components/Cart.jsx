import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import CheckOutCard from './CheckOutCard';
import { removeFromCart, checkout } from '../actions';

class Cart extends Component {
  onCheckoutClick = async () => {
    const { cart, checkout: placeOrder, history } = this.props;
    const foodItems = Object.keys(cart).map(Number);
    await placeOrder(foodItems);
    const {
      fetching: { fetching, errorMessage },
    } = this.props;
    if (!fetching && !errorMessage) return history.push('/order-history');
    return undefined;
  };

  onRemoveClick = (foodId) => {
    const { removeFromCart: removeItemFromCart } = this.props;
    removeItemFromCart(foodId);
  };

  render() {
    const { cart } = this.props;

    const cartItems = Object.keys(cart)
      .map(Number)
      .map(item => ({
        id: item,
        foodName: cart[item].foodName,
        foodPrice: cart[item].foodPrice,
      }));

    return (
      <div className="wrapper">
        <h2>Items To Be Purchased</h2>
        {cartItems.length ? (
          <Fragment>
            <section className="container checkout">
              {cartItems.map(item => (
                <CheckOutCard
                  key={item.id}
                  id={item.id}
                  foodName={item.foodName}
                  foodPrice={item.foodPrice}
                  buttonCallback={() => this.onRemoveClick(item.id)}
                />
              ))}
            </section>
            <div className="order-price">
              <p>
                <strong>total price</strong>
: â‚¦
                <span id="order-price">
                  {cartItems
                    .map(item => item.foodPrice)
                    .reduce((a, b) => a + b)
                    .toLocaleString()}
                </span>
              </p>
            </div>
            <button
              type="button"
              className="btn-primary"
              id="checkout"
              onClick={this.onCheckoutClick}
            >
              Checkout
            </button>
          </Fragment>
        ) : (
          <section className="container checkout">
            <div>No Items in cart</div>
          </section>
        )}
      </div>
    );
  }
}

Cart.propTypes = {
  cart: PropTypes.instanceOf(Object).isRequired,
  removeFromCart: PropTypes.func.isRequired,
  checkout: PropTypes.func.isRequired,
  history: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
  fetching: PropTypes.instanceOf(Object).isRequired,
};

const mapStateToProps = state => ({
  cart: state.cart,
  fetching: state.fetching,
});

export default connect(
  mapStateToProps,
  { removeFromCart, checkout },
)(Cart);
