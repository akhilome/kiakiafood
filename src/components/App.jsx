import '../index.css';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import propTypes from 'prop-types';
import { checkAuthStatus } from '../actions';
import HomePage from './HomePage';
import LoginPage from './LoginPage';
import Signup from './SignupPage';
import NotFoundPage from './NotFoundPage';

export class App extends Component {
  componentDidMount() {
    const { checkAuthStatus: checkUserAuthStatus } = this.props;
    checkUserAuthStatus();
  }

  render() {
    return (
      <Router>
        <Switch>
          <Route path="/" exact component={HomePage} />
          <Route path="/login" component={LoginPage} />
          <Route path="/signup" component={Signup} />
          <Route component={NotFoundPage} />
        </Switch>
      </Router>
    );
  }
}

App.propTypes = {
  checkAuthStatus: propTypes.func.isRequired,
};

export default connect(
  null,
  { checkAuthStatus },
)(App);
