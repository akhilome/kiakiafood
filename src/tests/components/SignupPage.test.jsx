import React from 'react';
import { shallow } from 'enzyme';

import { SignupPage } from '../../components/SignupPage';
import { mountWrap } from '../helpers/contextWrapper';
import { updateInput } from '../helpers';

describe('SignupPage Component', () => {
  const props = {
    signUpUser: jest.fn(),
    history: jest.fn(),
    isLoggedIn: false,
  };

  const altProps = {
    signUpUser: jest.fn(),
    history: { push: jest.fn() },
    isLoggedIn: true,
    fetching: false,
  };

  it('should render signup page correctly', () => {
    const wrapper = shallow(<SignupPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow user to fill signup form', () => {
    const wrapper = shallow(<SignupPage {...props} />);
    const form = wrapper.find('form');
    updateInput(form, 'input[name="name"]', 'kizito');
    updateInput(form, 'input[name="email"]', 'kay@gmail.com');
    updateInput(form, 'input[name="password"]', 'secretpassword');
    updateInput(form, 'input[name="password-confirm"]', 'secretpassword');
    updateInput(form, 'input[name="admin-secret"]', 'adminsecret');

    expect(wrapper.state('name')).toEqual('kizito');
    expect(wrapper.state('email')).toEqual('kay@gmail.com');
    expect(wrapper.state('password')).toEqual('secretpassword');
    expect(wrapper.state('confirmPassword')).toEqual('secretpassword');
    expect(wrapper.state('adminSecret')).toEqual('adminsecret');
  });

  it('should correct submit form with user details', () => {
    const wrapper = shallow(<SignupPage {...props} />);
    wrapper.find('form').simulate('submit', { preventDefault: () => '' });
    expect(props.signUpUser).toHaveBeenCalled();
  });

  it('should redirect user after signup', () => {
    const wrapper = shallow(<SignupPage {...altProps} />);
    wrapper.find('form').simulate('submit', { preventDefault: () => '' });
    expect(altProps.history.push).toHaveBeenCalled();
  });

  it('should redirect user if already logged in', () => {
    mountWrap(<SignupPage {...altProps} />);
    expect(altProps.history.push).toHaveBeenCalledWith('/menu');
  });
});
