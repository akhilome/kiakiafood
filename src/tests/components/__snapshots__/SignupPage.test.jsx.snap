// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage Component should render signup page correctly 1`] = `
<div
  className="wrapper auth"
>
  <section
    className="container"
  >
    <h2>
      Create An Account
    </h2>
    <form
      onSubmit={[Function]}
    >
      <input
        name="name"
        onChange={[Function]}
        placeholder="Your full name"
        required={true}
        type="text"
        value=""
      />
      <input
        name="email"
        onChange={[Function]}
        placeholder="Your Email Address"
        required={true}
        type="email"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Choose Password"
        required={true}
        type="password"
        value=""
      />
      <input
        name="password-confirm"
        onChange={[Function]}
        placeholder="Confirm Password"
        required={true}
        type="password"
        value=""
      />
      <input
        name="admin-secret"
        onChange={[Function]}
        placeholder="Admin Secret [optional]"
        type="password"
        value=""
      />
      <input
        className="btn-primary"
        name="submit"
        type="submit"
        value="Sign Up"
      />
    </form>
    <div>
      <p>
        Already have an account?
         
        <Link
          replace={false}
          to="/login"
        >
          Sign In
        </Link>
      </p>
    </div>
  </section>
</div>
`;
